### PR TITLE
gh-103092: Isolate winsound

### PR DESCRIPTION
**Pull request by aisk**
* Issue: gh-103092

### CODE REVIEW

File: PC/winsound.c [hunk 1]
```diff
@@ -254,5 +241,29 @@ PyInit_winsound(void)
...
+static struct PyModuleDef winsoundmodule = {
+    PyModuleDef_HEAD_INIT,
+    "winsound",
+    sound_module_doc,
+    0,
+    sound_methods,
+    sound_slots,
+    NULL,
+    NULL,
+    NULL
```
Comment (arhadthedev) [inline PC/winsound.c]: A useful feature of C99:

```suggestion
    .m_base = PyModuleDef_HEAD_INIT,
    .m_name = "winsound",
    .m_doc = sound_module_doc,
    .m_methods = sound_methods,
    .m_slots = sound_slots
```

File: Misc/NEWS.d/next/Library/2023-04-04-21-27-51.gh-issue-103092.7s7Bzf.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Adapt :mod:`!winsound` to :pep:`687`.
```
Comment (erlend-aasland) [inline Misc/NEWS.d/next/Library/2023-04-04-21-27-51.gh-issue-103092.7s7Bzf.rst]:
```suggestion
Adapt the :mod:`winsound` extension module to :pep:`687`.
```

File: PC/winsound.c [hunk 3]
```diff
@@ -254,5 +241,25 @@ PyInit_winsound(void)
...
+    {Py_mod_exec, exec_module},
+    {0, NULL}
+};
+
+static struct PyModuleDef winsoundmodule = {
+    .m_base = PyModuleDef_HEAD_INIT,
+    .m_name = "winsound",
+    .m_doc = sound_module_doc,
+    .m_methods = sound_methods,
+    .m_slots = sound_slots
```
Comment (erlend-aasland) [inline PC/winsound.c]:
```suggestion
    .m_slots = sound_slots,
```

**Comment by erlend-aasland** [comment]
I'd like to land gh-103385 first.

**Comment by erlend-aasland** [comment]
Please pull in `main`[^1], now that gh-103385 has landed.

[^1]: `git merge --no-ff main`; please do not rebase, as per the devguide.

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
@zooba, do you want to take a look?

**Pull request merged**